src/mlpack/methods/ann/layer/padding.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file padding.hpp
+ * @author Saksham Bansal
+ *
+ * Definition of the Padding class that pads the incoming data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_PADDING_HPP
+#define MLPACK_METHODS_ANN_LAYER_PADDING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/layer/layer_traits.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Padding module class. The Padding module applies a bias term
+ * to the incoming data.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class Padding
+{
+ public:
+  /**
+   * Create the Padding object using the specified number of output units.
+   *
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   */
+  Padding(const size_t padW = 0,
+          const size_t padH = 0);
```

Comment:
The layer looks good to me. Can you just add one more option which will take `valid` or `same` as an option and then pad accordingly?

Reply:
I think that `Same` and `valid` are options of the convolutional layers. The padding layer has nothing to do with it. The dimensions cannot be the `same` after padding.
I think it would be easier to implement this in a separate PR after this is merged.

Reply:
No, I don't think so. For having `same` padding `TensorFlow` uses a trick. 
Here are the links you can refer to.
https://www.tensorflow.org/api_docs/python/tf/pad

For `same` padding to make always possible:
https://stackoverflow.com/questions/37674306/what-is-the-difference-between-same-and-valid-padding-in-tf-nn-max-pool-of-t

For the same padding, if the rows to come out `odd` instead of `even`, then they add an extra row or column at the end. So if you wish to implement the way it is implemented in the first link you just will need to add a vector of p[n, 2] as an argument where n is the dimension and p[i, 0] specifies the pad to add at start and p[i, 1] specifies the pad to add at the end of the dimension.

Or what you do is just take `valid` and `same` as an argument and handle everything inside the constructor. It's totally up to you.
But yes to make some modification for sure.

Reply:
> Or what you do is just take valid and same as an argument and handle everything inside the constructor. It's totally up to you.

No, this will not work. The dimensions cannot be the same after padding so, it is illogical to pass the argument `same` to the Padding constructor.

> For the same padding, if the rows to come out odd instead of even, then they add an extra row or column at the end.

Alright, I think this could be done. Basically, you want flexibility to have different padding on different sides (to allow for odd padding).

Reply:
Yes